### ISSUE TITLE
Lower cased globalFqn property in some ExternalModule and ExternalDeclaration nodes

### ISSUE THREAD

**Issue by JohT** [jqassistant-plugin/jqassistant-typescript-plugin#34]
Hi 👋

I've discovered that some `ExternalModule` and `ExternalDeclaration` nodes have a lower-cased `globalFqn` property. 
Is that intended or a technical necessity? Or might that be a bug?

There is for example an `ExternalModule` with lower-cased `globalFqn` = `/users/johnny/repositories/git/react-router/packages/react-router/lib/context.ts`. Another one has the original (unchanged) `globalFqn` = `/Users/johnny/Repositories/git/react-router/packages/router/router.ts`. (see Example Query 1)

Nodes labelled with `ExternalModule` and `ExternalDeclaration` are the only ones that are affected by this as far as i've seen. 

**Edit:** As it seems this also applies to `ExternalType` nodes. Some of them have a lower-cased property `ts.referencedGlobalFqn` like `"/users/johnny/repositories/git/react-router/packages/react-router/lib/context.ts".RouteObject"`, some not like "/users/johnny/repositories/git/react-router/packages/react-router/lib/context.ts".RouteObject". (see Example Query 2)

### Example Query

```cypher
 MATCH (ts:TS)
 WHERE ts.globalFqn IS NOT NULL
RETURN labels(ts)
      ,split(ts.globalFqn, '/')[1] AS globalRoot
      ,count(*) as nodeCount
      ,collect(ts.globalFqn)[0..4] AS exampleGlobalFqns
```

### Example Query 2

```cypher
 MATCH (ts:TS)
  WITH ts, split(ts.referencedGlobalFqn, '/')[1] AS globalRoot
 WHERE globalRoot IS NOT NULL
RETURN labels(ts)
      ,globalRoot
      ,count(*) as nodeCount
      ,collect(ts.referencedGlobalFqn)[0..4] AS exampleReferencedGlobalFqns
```

### Example data

Taken from [react-router](https://github.com/remix-run/react-router): [ts-output-react-router.json](https://github.com/jqassistant-plugin/jqassistant-typescript-plugin/files/14764231/ts-output.json)

**Comment by SebastianWendorf** [comment]
This is not intentional behavior. The capitalization should be preserved for all paths.

**Comment by JohT** [comment]
Yes, I'm using MacOS.

**Comment by SebastianWendorf** [comment]
Hi. Yes, you can try it on Windows. My guess is, that it should work there.
The problem is, that file paths on MacOS aren't normalized with respect to case-preservation like the Windows ones. I will change this behavior, however as I can't test it, I cannot guarantee a fix.

**Comment by JohT** [comment]
You were most likely right with your presumption that it might be related to the file system or MacOS. I couldn't reproduce the problem on Windows. 

I've also collected some test errors in case these would be of any help for you.
If you want I can check out a branch and run the unit tests on my machine if you want to try something out.
What you also can do is add another GitHub pipeline that runs on Mac as long as it doesn't lead to any costs (which shouldn't be the case for open source projects).

### MacOs

I had a look at the json that i've got from the scan on MacOs. Here is a small excerpt of it:
```json
"dependency": [
                {
                    "globalFqn": "\"/users/johnny/repositories/git/react-router/packages/react-router/lib/context.ts\".RouteObject",
                    "localFqn": "",
                    "targetType": "declaration",
                    "globalSourceFQN": "\"/Users/johnny/Repositories/git/react-router/packages/react-router\".mapRouteProperties",
                    "sourceType": "declaration",
                    "cardinality": 1
                }
```

### Windows

I couldn't reproduce it on windows as you presumed:

```json
"dependency": [
                {
                    "globalFqn": "\"G:/Eigene Dateien/Engineering/Repositories/git/react-router/packages/react-router/lib/context.ts\".RouteObject",
                    "localFqn": "",
                    "targetType": "declaration",
                    "globalSourceFQN": "\"G:/Eigene Dateien/Engineering/Repositories/git/react-router/packages/react-router\".mapRouteProperties",
                    "sourceType": "declaration",
                    "cardinality": 1
                }
```

### Failing Tests

I've tried to check out the project and run its unit tests on my Mac and several of them are failing. I'm not sure if this is related to this issue or if I've missed some mandatory setup steps though....

Example:

```
 ● Multi-Project test › basic project (project1)

    expect(received).toHaveLength(expected)

    Expected length: 0
    Received length: 1
    Received array:  [{"declarations": [{"globalFqn": "\"/users/johnny/repositories/git/jqassistant-typescript-plugin/typescript/test/core/integration/sample-projects/multi-project/project1/src/module2.ts\".Project1Class", "localFqn": "", "name": "Project1Class"}], "globalFqn": "/users/johnny/repositories/git/jqassistant-typescript-plugin/typescript/test/core/integration/sample-projects/multi-project/project1/src/module2.ts", "localFqn": ""}]

      36 |             const externalModules = project.concepts.get(LCEExternalModule.conceptId);
      37 |             expect(externalModules).toBeDefined();
    > 38 |             expect(externalModules).toHaveLength(0);
         |                                     ^
      39 |
      40 |             const [projectRootPath,
      41 |                 modules,

      at Object.<anonymous> (test/core/integration/multi-project.test.ts:38:37)

  ● Multi-Project test › sibling project using a common reference project (project2)

    expect(received).toHaveLength(expected)

    Expected length: 0
    Received length: 1
    Received array:  [{"declarations": [{"globalFqn": "\"/users/johnny/repositories/git/jqassistant-typescript-plugin/typescript/test/core/integration/sample-projects/multi-project/project2/src/module2.ts\".Project2Class", "localFqn": "", "name": "Project2Class"}], "globalFqn": "/users/johnny/repositories/git/jqassistant-typescript-plugin/typescript/test/core/integration/sample-projects/multi-project/project2/src/module2.ts", "localFqn": ""}]

      72 |             const externalModules = project.concepts.get(LCEExternalModule.conceptId);
      73 |             expect(externalModules).toBeDefined();
    > 74 |             expect(externalModules).toHaveLength(0);
         |                                     ^
      75 |
      76 |             const [projectRootPath,
      77 |                 modules,

      at Object.<anonymous> (test/core/integration/multi-project.test.ts:74:37)
```

**Comment by SebastianWendorf** [comment]
I'll check the option to run the typescript tests on multiple platforms

**Comment by SebastianWendorf** [comment]
New version (1.1.0) of the npm package is released that should fix the incorrect behavior on MacOS.

**Comment by JohT** [comment]
Thank you very much @SebastianWendorf 👍. Looks good on my side 👏.
- I was able to run `npm run test` successfully (local clone on main branch at [a80edd1](https://github.com/jqassistant-plugin/jqassistant-typescript-plugin/commit/a80edd14b8e6082052d1466476effa1aa8f57760)) on my Mac.
- I also checked the resulting JSON and couldn't find any lower cased file paths anymore.
- I've now also merged my first pull request for fully automated Typescript  report generation based on jQAssistant: https://github.com/JohT/code-graph-analysis-pipeline .I'll continue working on that to get it close to what is there for Java. Even though I only have a pipeline for react router as an example, your plugin is now running every time my pipeline runs :-) .
